### PR TITLE
fread into readTableFast and added reading zip files

### PR DESCRIPTION
**Pull request by katwre**

### CODE REVIEW

File: R/readData.R [hunk 1]
```diff
@@ -1,19 +1,91 @@
...
+    if(grepl("^.*(.gz)[[:space:]]*$", filename)){
+      filename <- paste("gzip -dc", filename)
+    } else if (grepl("^.*(.zip)[[:space:]]*$", filename)){
+      filename <- paste("unzip -p", filename)
+    } else if (grepl("^.*(.bz2)[[:space:]]*$", filename)){  
+      filename <- paste("bzcat -dc", filename)
+    } 
+    
+    # removes the track header 
+    if(grepl("^.*(.gz|.bz2|.zip)[[:space:]]*$", filename)){
```
Comment (al2na) [inline R/readData.R]: it would be better if track header checking would stay as part of readGeneric not readTableFast

File: R/readData.R [hunk 2]
```diff
@@ -73,21 +145,9 @@ readTableFast<-function(filename,header=TRUE,skip=0,sep=""){
 #' @rdname readGeneric
 readGeneric<-function(file, chr=1,start=2,end=3,strand=NULL,meta.cols=NULL, 
                       keep.all.metadata=FALSE, zero.based=FALSE, 
-                      remove.unusual=FALSE, header=FALSE, 
-                      skip=0, sep="\t"){
+                      remove.unusual=FALSE, header="auto", 
+                      skip=0, sep="auto"){
```
Comment (al2na) [inline R/readData.R]: why is header and sep "auto' ? If you are changing these you have to change the documentation (function help pages ) as well. My suggestion is to leave them as they are. The "auto" option is not compatible with current help pages or people who are used to base::read.table

**Comment by al2na** [comment]
see the comments above, please make necessary changes and update

File: R/readData.R [hunk 3]
```diff
@@ -1,19 +1,91 @@
 # ---------------------------------------------------------------------------- #
 # fast reading of big tables
-readTableFast<-function(filename,header=TRUE,skip=0,sep=""){
+readTableFast<-function(filename,header="auto",skip=0,sep="auto",as.datatable=FALSE, ...){
```
Comment (al2na) [inline R/readData.R]: header and sep here can stay as "auto", this function is not exported

File: R/readData.R [hunk 4]
```diff
@@ -1,19 +1,91 @@
...
+    
+    # default header and sep for read.table
+    if(header=="auto") header=FALSE
+    if(sep=="auto") sep=""
+    
+    # if file is not in zip format
+    if (!grepl("^.*(.zip)[[:space:]]*$", filename)){
+      
+      # removes the track header
+      track=scan(filename, n=1, what='character', sep='\n', quiet=TRUE)
```
Comment (al2na) [inline R/readData.R]: again let's decide if the file has track header or not outside this function

File: R/readData.R [hunk 5]
```diff
@@ -1,19 +1,91 @@
...
+    
+    # if file is not in zip format
+    if (!grepl("^.*(.zip)[[:space:]]*$", filename)){
+      
+      # removes the track header
+      track=scan(filename, n=1, what='character', sep='\n', quiet=TRUE)                    
+      if(grepl('^>',track))
+        skip = max(1, skip)
+      # if header = FALSE, checks whether a header line exists, and skips it
+      if(header==FALSE){
```
Comment (al2na) [inline R/readData.R]: I think this should be header=="auto, to be on the safe side. if the user sets header=FALSE, then we try to find a header, we will find a header if all the elements of the table is made of characters. This supposed to be a generic function, should work on all tables, not only bed-like tables

File: R/readData.R [hunk 6]
```diff
@@ -1,19 +1,91 @@
...
+      skip = max(1, skip)
+    
+    df = fread(filename, 
+               header=header, skip=skip,
+               sep=sep, stringsAsFactors=FALSE,
+               data.table=as.datatable)
+  }else{
+    
+    # default header and sep for read.table
+    if(header=="auto") header=FALSE
```
Comment (al2na) [inline R/readData.R]: see below, this is a bit dangerous

File: R/readData.R [hunk 7]
```diff
@@ -73,21 +145,9 @@ readTableFast<-function(filename,header=TRUE,skip=0,sep=""){
 #' @rdname readGeneric
 readGeneric<-function(file, chr=1,start=2,end=3,strand=NULL,meta.cols=NULL, 
                       keep.all.metadata=FALSE, zero.based=FALSE, 
-                      remove.unusual=FALSE, header=FALSE, 
-                      skip=0, sep="\t"){
+                      remove.unusual=FALSE, header="auto", 
+                      skip=0, sep="auto"){
 
-  # removes the track header
-  track=scan(file, n=1, what='character', sep='\n', quiet=TRUE)
```
Comment (al2na) [inline R/readData.R]: using scan() like this on a gzipped file is quite fast.

**Comment by al2na** [comment]
are you also doing full R  CMD Check before submitting ?

**Comment by katwre** [comment]
sorry, this time I did R CMD Check before pushing code here

**Pull request merged**